### PR TITLE
LPS-125039 Make sure that available languages are consistent across all the localizable properties

### PR DESCRIPTION
**Pull request by pavel-savinov**

### CODE REVIEW

**Comment by pavel-savinov** [comment]
ci:test:sf

**Comment by pavel-savinov** [comment]
ci:test:relevant

**Comment by jeyvison** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/DDMFormJSONSerializer.java [hunk 1]
```diff
@@ -66,9 +67,31 @@ public DDMFormSerializerSerializeResponse serialize(
 				ddmForm.getDefinitionSchemaVersion());
 		}
 
+		List<DDMFormField> ddmFormFields = ddmForm.getDDMFormFields();
+
+		for (DDMFormField ddmFormField : ddmFormFields) {
```
Comment (jeyvison) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/DDMFormJSONSerializer.java:72]: Hi @pavel-savinov , Can you please sort this? Just put 
`_addMissingLocales(
				ddmFormField.getPredefinedValue(),
				ddmForm.getAvailableLocales());`
between 
`...ddmFormField.getLabel...` and `...ddmFormField.getTips()...`

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/DDMFormJSONSerializer.java [hunk 2]
```diff
@@ -164,6 +187,22 @@ protected JSONObject toJSONObject(LocalizedValue localizedValue) {
 		return jsonObject;
 	}
 
+	private void _addMissingLocales(
+		LocalizedValue localizedValue, Set<Locale> availableLocales) {
+
+		String defaultLabel = localizedValue.getString(
```
Comment (jeyvison) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/DDMFormJSONSerializer.java:193]: Can you please change the variable name to `defaultLocaleValue` ?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/DDMFormJSONSerializer.java [hunk 3]
```diff
@@ -164,6 +187,22 @@ protected JSONObject toJSONObject(LocalizedValue localizedValue) {
...
+			localizedValue.getDefaultLocale());
+
+		Set<Locale> valueAvailableLocales =
```
Comment (jeyvison) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/DDMFormJSONSerializer.java:196]: Can you please change the variable name to `localizedValueAvalilableLocales`?

**Review by jeyvison — CHANGES_REQUESTED** [review]
Hi @pavel-savinov , please , see my comments. Thanks :)

**Comment by pavel-savinov** [comment]
@jeyvison Sorted/renamed! Thanks!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/DDMFormJSONSerializer.java [hunk 4]
```diff
@@ -164,6 +187,22 @@ protected JSONObject toJSONObject(LocalizedValue localizedValue) {
 		return jsonObject;
 	}
 
+	private void _addMissingLocales(
+		LocalizedValue localizedValue, Set<Locale> availableLocales) {
+
+		String defaultLocaleValue = localizedValue.getString(
+			localizedValue.getDefaultLocale());
+
+		Set<Locale> localizedValueAvalilableLocales =
```
Comment (jeyvison) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/DDMFormJSONSerializer.java:196]: Hi @pavel-savinov . Can you please fix this typo? Thanks!

**Review by jeyvison — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/DDMFormJSONSerializer.java (hunk 4 above)
Comment (pavel-savinov) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/io/DDMFormJSONSerializer.java]: @jeyvison fixed :)

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by pavel-savinov** [comment]
ci:test:relevant

**Comment by jeyvison** [comment]
Hey @pavel-savinov , the same poshi test failed two times. Can you please take a look if the you changes affected it?

**Comment by pavel-savinov** [comment]
ci:forward

**Comment by pavel-savinov** [comment]
@jeyvison Tests failures are related to https://issues.liferay.com/browse/LRQA-64305, not to this particular PR.

**Comment by jeyvison** [comment]
ci:forward

**Review by jeyvison — APPROVED** [review]

**Pull request closed without merging**